instrumentation/rw_test.go

```diff
@@ -0,0 +1,14 @@
+package instrumentation
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/fd/httpmiddlewarevet"
```

Comment:
This is why @romainmenke and I made httpmiddlewarevet. to check, test and catch any middleware that fails to conform to the complete ResponseWriter interface.

Do note that the feature set of the underlying ResponseWriter is preserved. In other words Pusher is only available when H2 is used and Hijacker is only available when H2 is not used.

When a ResponseWriter doesn't conform to the complete interface an error will be reported in the tests.

Wrapping ResponseWriters is more complicated than it might seem at first.
For instance http.Hijacker and io.ReaderFrom are only available for HTTP/HTTPS.
And http.Pusher is obviously only available for H2.

We have been using this tool to test all the middleware we write (or vendor) to make sure our ResponseWriter remains feature complete.